### PR TITLE
headphones-toolbox: init at 0.0.3

### PR DESCRIPTION
**Pull request by lambdachad**
## Description of changes

This PR adds the headphones-toolbox package, which is a UI for configuring Ploopy Headphones made with Tauri. Note that this doesn't add the required udev rules which must exist to make it work: https://github.com/george-norton/headphones-toolbox#quickstart-guide

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Nyabinary** [comment]
I fully tested it with my Ploopy headphones and it works great, I tested filters and firmware updates, both of them worked.

File: pkgs/by-name/he/headphones-toolbox/package.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  version = "0.0.3";
+
+  src = fetchurl {
+    url = "https://github.com/george-norton/headphones-toolbox/releases/download/headphones-toolbox-beta-v4r2/ploopy-headphones-toolbox_${finalAttrs.version}_amd64.deb";
+    hash = "sha256-r+ybcD6koSIJ/6cck3RNXmf758sRnhS1Y4kaYCNbveA=";
+  };
+
+  unpackCmd = ''
+    dpkg-deb -x $curSrc source
+  '';
```
Comment (wegank) [inline pkgs/by-name/he/headphones-toolbox/package.nix:20]:
```suggestion
```

File: pkgs/by-name/he/headphones-toolbox/package.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    webkitgtk
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
```
Comment (wegank) [inline pkgs/by-name/he/headphones-toolbox/package.nix:33]:
```suggestion
    runHook preInstall

    mkdir -p $out/bin
```

**Comment by wegank** [comment]
Please squash the last three commits, thanks!

**Comment by knarkzel** [comment]
> Please squash the last three commits, thanks!

Fixed.

**Pull request merged**